src/Tasks.UnitTests/XslTransformation_Tests.cs

```diff
@@ -848,6 +863,57 @@ public void XsltDocumentThrowsError()
             CleanUp(dir);
         }
 
+        /// <summary>
+        /// Xslt PreserveWhitespace = true
+        /// </summary>
+        [Fact]
+        public void XsltPreserveWhitespace()
+        {
+            string dir;
+            TaskItem[] xmlPaths;
+            TaskItem xslPath;
+            TaskItem[] outputPaths;
+            MockEngine engine;
+            Prepare(out dir, out xmlPaths, out xslPath, out _, out outputPaths, out _, out _, out engine);
+
+            // load transformed xsl and assert it is well formatted
+            {
+                XslTransformation t = new XslTransformation();
+                
+                t.BuildEngine = engine;
+                t.XslInputPath = xslPath;
+                t.XmlInputPaths = xmlPaths;
+                t.OutputPaths = outputPaths;
+                t.Parameters = _xslParameters;
+                t.PreserveWhitespace = true;
+
+                t.Execute();
+                Console.WriteLine(engine.Log);
+                using (StreamReader sr = new StreamReader(t.OutputPaths[0].ItemSpec))
+                {
+                    string fileContents = sr.ReadToEnd();
+                    Assert.Equal(
+@"<?xml version=""1.0"" encoding=""utf-8""?>
```

Comment:
To make this a more robust test, maybe make weird whitespace changes in the input and output files?

Reply:
I've tried to add another xml document with extra whitespaces
```csharp
        private readonly string _xmlDocument3 =
@"<root
    Name=""param1""
    Value=""value111"">

    <abc>

        <cde/>


    </abc>

</root>";
```
But toggling `PreserveWhitespace` from true to false doesn't change the output.
It's always the same
```xml
<?xml version="1.0" encoding="utf-8"?>
<surround>
  <root>
    <surround Name="param1" />
    <surround Value="value111" />
    <surround>

    </surround>
    <surround>
      <abc>
        <surround>

        </surround>
        <surround>
          <cde />
        </surround>
        <surround>


    </surround>
      </abc>
    </surround>
    <surround>

</surround>
  </root>
</surround>
```

Maybe only certain xsl transformations mess up the whitespaces. I've tried to look up the [xsl file](https://github.com/NMertsch/msbuild-xsltransformation-issue/blob/main/xsl/fdl2proto.xsl) used in the project where this issue happens but it was too complicated to add as a test case, and I think it imports some libraries or something.

Reply:
I think the surround tags are just what the xsl document, which is provided in the Unit Test file, does.
```csharp
private readonly string _xslDocument = "<xsl:stylesheet version=\"1.0\" xmlns:xsl=\"http://www.w3.org/1999/XSL/Transform\" xmlns:msxsl=\"urn:schemas-microsoft-com:xslt\" exclude-result-prefixes=\"msxsl\"><xsl:output method=\"xml\" indent=\"yes\"/><xsl:template match=\"@* | node()\"><surround><xsl:copy><xsl:apply-templates select=\"@* | node()\"/></xsl:copy></surround></xsl:template></xsl:stylesheet>";
 ```
 I haven't worked with xsl files before, so I'm not sure. Can you link some docs about them so that I can understand the [xsl file](https://github.com/NMertsch/msbuild-xsltransformation-issue/blob/main/xsl/fdl2proto.xsl) better? I think the transformation mentioned in the [issue](https://github.com/dotnet/msbuild/issues/7671) is about `fdl2proto`, not sure what are these tho.

Reply:
I've added the documents used [here](https://github.com/NMertsch/msbuild-xsltransformation-issue) and used it in the test. Now toggling the `PreserveWhitespace` makes the test fail or pass.